backend/routes/recommended/index.js

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+const router = require('express').Router()
+
+/* Import DB */
+const usersInfo = require('../../userInfo.json')
+const likeHistory = require('../../likeHistory.json')
+
+/* Successfully inquiry of authentication */
+router.get('/', (req, res) => {
+  const userId = req.query.userId //userIdはクエリパラメータで取得
+  res.status(200)
+
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+  const myLiked = likeHistory[userId]
+  const matchedUserInfo = {}
+  const unMatchedUserInfo = {}
+
+  //マッチングユーザ抽出
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+  Object.keys(myLiked).forEach((key1) => {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-member-access
+    Object.keys(likeHistory[myLiked[key1]]).forEach((key2) => {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      if (likeHistory[myLiked[key1]][key2] === userId) {
```

Comment:
下記でも同様の結果になる気がしますが、いかがでしょうか？
それか、JSの組み込み関数で配列の要素が一致しているかを探索する関数もあるので、
そちらを使用した方が意図が明確になるかもしれません🙇

https://www.javadrive.jp/javascript/array/index13.html

```suggestion
     if (key2 === userId) {
```

Reply:
@kan-mayoshi 
likeHistoryが下記構成なのでご提案だと動作が違ってしまいます・・・
読みにくいですが、暫定的にそのままとさせてください。
`{
  "3f328652-f4bb-4254-972a-d70489794a25": ["e857624e-ace4-44a5-8c9f-b9203f10df1f", "8c978007-dd53-428e-a486-12cb378564f8"],
  "e857624e-ace4-44a5-8c9f-b9203f10df1f": ["3f328652-f4bb-4254-972a-d70489794a25", "fc08e44e-befb-425e-a6c7-c6f4b2a57801", "ff85c29f-5522-4f4d-b4e1-da08e1311e3d"],
  "b830fcc6-b691-462a-beb0-20a73eeed2d9": ["e857624e-ace4-44a5-8c9f-b9203f10df1f"],
  "34edabec-417b-48db-a6a5-684ccd94315a": ["3f328652-f4bb-4254-972a-d70489794a25"],
  "24369dee-ac2f-442c-a39a-7f7139d72024": ["3f328652-f4bb-4254-972a-d70489794a25"],
  "ff85c29f-5522-4f4d-b4e1-da08e1311e3d": ["e857624e-ace4-44a5-8c9f-b9203f10df1f"],
  "09d94792-b443-4b8c-af17-959758de7aea": ["3f328652-f4bb-4254-972a-d70489794a25"],
  "8c978007-dd53-428e-a486-12cb378564f8": ["3f328652-f4bb-4254-972a-d70489794a25"],
  "fc08e44e-befb-425e-a6c7-c6f4b2a57801": ["e857624e-ace4-44a5-8c9f-b9203f10df1f"]
}`